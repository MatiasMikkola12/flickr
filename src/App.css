.data-wrapper {
  display: flex;
  flex-wrap: wrap;
}

.item-wrapper {
  background-color: lightgray;
  margin: 16px;
  border: 1px solid black;
  max-width: 250px;
  min-height: 350px;
}

.contents {
  margin: 8px;

}

img {
  width: 100%;
  height: 150px;
}

.title-wrapper {
  display: flex;
  margin-bottom: 16px;
}

.description-wrapper {
  overflow: hidden;
  text-overflow: ellipsis;
}

.dropdown-wrapper {
  display: flex;
  justify-content: center;
}

select, option {
  width: 100px;
}