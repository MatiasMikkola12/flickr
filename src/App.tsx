import React from 'react';
import './App.css';

interface AppProps {
}

interface AppState {
  data: any
}

interface Item {
  title: string
  media: any
  author: string
  description: string
  link: string
  tags: string;
}

class App extends React.Component<AppProps, AppState> {
  constructor(props: any) {
    super(props);

    this.state = {
      data: [],
    }
  }

  componentDidMount() {
    fetch('data/data.json')
      .then((response) => {
        if (!response.ok) {
          throw new Error('Response not OK');
        }
        return response.json();
      })
      .then((data) => {
        this.setState({ data: data })
      })
      .catch((error) => {
        console.error('Some issue with the fetch:', error);
      });
  }

  render() {
    const items = this.state.data.items
    return (
      <div className="data-wrapper">
        {items && items.map((item: Item) => (
          <div className="item-wrapper">
            <div className="contents">
              <div className="image-wrapper">
                <img src={item.media.m} />
              </div>
              <div className="title-wrapper">
                <span><a href={item.media.m}>{item.title}</a> by <a href={item.link}>{item.author}</a></span>
              </div>

              <div className="description-wrapper">
                <span>Description: {item.description}</span>
              </div>

              <div className="dropdown-wrapper">
                <select className="dropdown" id="tags">
                  <option value="Tag">Tags</option>

                  {item.tags && item.tags.split(" ").map((tag: string) => (
                    <option value="tag">{tag}</option>
                  )
                  )}
                </select>
              </div>
            </div>
          </div>
        ))}
      </div>
    )
  }
}

export default App;
