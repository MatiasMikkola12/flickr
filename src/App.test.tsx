import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('snapshot of the contents of the card', () => {
  const theapp = renderer.create(<App />).toJSON();
  expect(theapp).toMatchSnapshot();
});
