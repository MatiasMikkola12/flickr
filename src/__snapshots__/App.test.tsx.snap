// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the contents of the card 1`] = `
<div
  className="data-wrapper"
/>
`;
